pkg/docker/events.go

```diff
@@ -193,3 +204,43 @@ func validatePortMapping(portMap nat.PortMap) {
 		}
 	}
 }
+
+// When the port binding is bound to 127.0.0.1, we add an additional DNAT rule in the main
+// DOCKER chain after the existing rule (using --append).
+// This is necessary because the initial DOCKER DNAT rule that is created by docker only allows
+// the traffic to be routed to localhost from localhost. Therefore, we add an additional rule to
+// allow traffic to any destination IP address which allows the service to be discoverable through
+// namespaced network's subnet.
+// This is required as the traffic is routed via vm-switch over the tap network.
+// The existing DNAT rule are as follows:
+// DNAT       tcp  --  anywhere             localhost            tcp dpt:9119 to:10.4.0.22:80.
+// We enter the following rule after the existing rule:
+// DNAT       tcp  --  anywhere             anywhere             tcp dpt:9119 to:10.4.0.22:80.
+func createLoopbackIPtablesRules(containerIP string, portMappings nat.PortMap) error {
+	var errs []error
+
+	for portProto, portBindings := range portMappings {
+		for _, portBinding := range portBindings {
+			if portBinding.HostIP == "127.0.0.1" {
+				//nolint:gosec // no security concern with the potentially tainted command arguments
+				iptableCmd := exec.Command("iptables",
```

Comment:
Perhaps we should fork `github.com/lima-vm/lima/pkg/guestagent/iptables` (which we already use) or push something upstream to do this for us?